lib/SlackBot.js

```diff
@@ -246,20 +246,21 @@ function Slackbot(configuration) {
                 // Recent changes in Slack will break other integrations as they no longer
                 // require a bot and therefore Slack won't send the bot information.
                 if (payload.type === 'event_callback') {
-
                     if (!team.bot) {
                         slack_botkit.log.error('No bot identity found.');
                         return res.status(500).send();
                     }
 
+                    bot.identity = {
+                        id: team.bot.user_id,
+                        name: team.bot.name
+                    };
+                } else {
+                    bot.identity = {
```

Comment:
Generated the original issue using a bot attached to a Slack app. Used same bot to create this fix. When sending a slash command to my bot, the `team` object was in the form `{ id: 'XXXXXXXXX', name: 'foo', domain: 'foo' }`, and there was no `payload.type` defined.

Based on my understanding of legacy versus Slack connected bots, I believe this approach should handle both types. Using this patch with my bot restores slash command functionality.